chain/manager.go

```diff
@@ -928,13 +928,23 @@ func (m *Manager) UnconfirmedParents(txn types.Transaction) []types.Transaction
 	return parents
 }
 
-// V2UnconfirmedParents returns the v2 transactions in the txpool that are referenced
-// by txn.
-func (m *Manager) V2UnconfirmedParents(txn types.V2Transaction) []types.V2Transaction {
+// V2TransactionSet returns the full transaction set and basis necessary for
+// broadcasting a transaction. If the provided basis does not match the current
+// tip the transaction will be updated. The transaction set includes the parents
+// and the transaction itself in an order valid for broadcasting.
+func (m *Manager) V2TransactionSet(basis types.ChainIndex, txn types.V2Transaction) (types.ChainIndex, []types.V2Transaction, error) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 	m.revalidatePool()
 
+	// update the transaction's basis to match tip
+	_, txns, err := m.updateV2TransactionSet(basis, []types.V2Transaction{txn})
+	if err != nil {
+		return types.ChainIndex{}, nil, fmt.Errorf("failed to update transaction set basis: %w", err)
+	}
+	txn = txns[0]
```

Comment:
Shouldn't we handle the case where `txns` comes back empty? If not I would still go for a sanity check and error out here.